wcfsetup/install/files/lib/core.functions.php

```diff
@@ -90,6 +90,97 @@ function wcfDebug() {
 	}
 }
 
+namespace wcf\functions\deprecated {
+	use \wcf\system\exception\IExtraInformationException;
+
+	function DeprecationError($message, $extraInformation) {
+		// The error needs to be defined lazily, because core.functions.php is initialized before the autoloader is ready.
+		if (!class_exists(DeprecationError::class)) {
+			class DeprecationError extends \Error implements IExtraInformationException {
+				public function __construct($message, $extraInformation) {
+					parent::__construct($message);
+					$this->extraInformation = $extraInformation;
+				}
+				
+				public function getExtraInformation() {
+					return $this->extraInformation;
+				}
+			}
+		}
+		
+		return new DeprecationError($message, $extraInformation);
+	}
+	
+	function feature($message, $extraInformation) {
+		if (!ENABLE_DEPRECATION_WARNINGS) return;
+		
+		throw DeprecationError("A deprecated feature is used, while 'ENABLE_DEPRECATION_WARNINGS' is enabled: ".$message, $extraInformation);
+	}
+	
+	/**
+	 * Sends a deprecation warning for the given class when the returned
+	 * object is destructed.
+	 * 
+	 * Usage:
+	 * 
+	 * $_ = \wcf\functions\deprecated\_class(Foo::class);
+	 * class Foo { ... }
+	 * 
+	 * @param	string	$className
+	 * @return	object
+	 */
+	function _class($className) {
+		if (!ENABLE_DEPRECATION_WARNINGS) return;
+
+		return new class($className) {
```

Comment:
Please add a comment explaining why this indirect way of throwing the exception is necessary.